config-ui/src/components/blueprints/create-workflow/DataTransformations.jsx

```diff
@@ -136,17 +109,47 @@ const DataTransformations = (props) => {
   )
   const [activeEntity, setActiveEntity] = useState()
 
-  const transformationHasChanged = useCallback(
+  const checkCurrentTransformationIsChanged = useCallback(
     (item) => {
-      const initializeTransform = initializeTransformations[item?.id]
-      const storedTransform = transformations[item?.id]
-      return (
-        initializeTransform &&
-        storedTransform &&
-        !isEqual(initializeTransform, storedTransform)
+      return checkTransformationIsChanged(
+        configuredConnection?.provider,
+        configuredConnection?.id,
+        item
       )
     },
-    [initializeTransformations, transformations]
+    [
+      configuredConnection?.provider,
+      configuredConnection?.id,
+      checkTransformationIsChanged
+    ]
+  )
+
+  const changeCurrentTransformationSettings = useCallback(
+    (settings) => {
```

Comment:
Is looks like we assume Board, then default to Project. It possible to make this more declarative?

```suggestion
    (settings) => {
        console.log(configuredBoard || configuredProject)
        return changeTransformationSettings(
          configuredConnection?.provider,
          configuredConnection?.id,
          configuredBoard || configuredProject,
          settings
        )
     }
```

Reply:
fixed